### PR TITLE
analysis: Fix plat_info uses

### PR DESCRIPTION
**Pull request by valschneider**

### CODE REVIEW

File: lisa/analysis/frequency.py [hunk 1]
```diff
@@ -308,7 +308,11 @@ def plot_cpu_frequencies(self, cpu, filepath=None, axis=None):
         if local_fig:
             fig, axis = self.setup_plot()
 
-        frequencies = self.trace.plat_info['freqs'][cpu]
+        try:
+            frequencies = self.trace.plat_info['freqs'][cpu]
```
Comment (douglas-raillard-arm) [inline lisa/analysis/frequency.py]: Could we split indexing on `['freqs']` (inside the `try`) and `[cpu]` (outside or in `else` clause) ?
This way we would still fail if a platform info is provided with incomplete frequencies, which would probably be the sign of a user error (if somebody wants auto-detection, then one can omit the `freqs` keys altogether).

Comment (valschneider) [reply lisa/analysis/frequency.py]: Would this work for you then?

```python
        if "freqs" in self.trace.plat_info:
            frequencies = self.trace.plat_info['freqs'][cpu]
        else:
            self.get_logger().info("Estimating CPU frequencies from trace")
            frequencies = df.frequency.unique()
```

Comment (douglas-raillard-arm) [reply lisa/analysis/frequency.py]: yes, that's equivalent (less efficient but nobody cares in that case)

**Comment by valschneider** [comment]
@douglas-raillard-arm Done

File: lisa/trace.py [hunk 2]
```diff
@@ -143,6 +143,7 @@ def cpus_count(self):
         # If we don't know the number of CPUs, check the trace for the
         # highest-numbered CPU that traced an event.
         except KeyError:
+            self.get_logger().info("Estimating CPU count from trace")
```
Comment (douglas-raillard-arm) [inline lisa/trace.py]: Can we log that after the estimation and show the estimated number as well ? That can speed up debugging if the log already contains the info straight away

File: lisa/analysis/frequency.py [hunk 3]
```diff
@@ -308,7 +308,11 @@ def plot_cpu_frequencies(self, cpu, filepath=None, axis=None):
         if local_fig:
             fig, axis = self.setup_plot()
 
-        frequencies = self.trace.plat_info['freqs'][cpu]
+        if "freqs" in self.trace.plat_info:
+            frequencies = self.trace.plat_info['freqs'][cpu]
+        else:
+            self.get_logger().info("Estimating CPU frequencies from trace")
```
Comment (douglas-raillard-arm) [inline lisa/analysis/frequency.py]: same as for the other logging

**Comment by valschneider** [comment]
@douglas-raillard-arm Done

**Pull request merged**